core/storage/question/gae_models_test.py

```diff
@@ -388,6 +389,49 @@ def test_get_question_skill_links_based_on_difficulty(self):
         self.assertTrue(questionskilllink_model2 in question_skill_links)
         self.assertTrue(questionskilllink_model4 in question_skill_links)
 
+    def test_get_random_question_skill_links_based_on_difficulty(self):
+        questionskilllink_model1 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id1', 'skill_id1', 0.6)
+            )
+        questionskilllink_model2 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id2', 'skill_id1', 0.6)
+            )
+        questionskilllink_model3 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id3', 'skill_id1', 0.6)
+            )
+        questionskilllink_model4 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id4', 'skill_id1', 0.6)
+            )
+        questionskilllink_model5 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id5', 'skill_id1', 0.6)
+            )
+        question_models.QuestionSkillLinkModel.put_multi_question_skill_links(
+            [questionskilllink_model1, questionskilllink_model2,
+             questionskilllink_model3, questionskilllink_model4,
+             questionskilllink_model5])
+        utils.random.seed(1)
+        question_skill_links_1 = (
+            question_models.QuestionSkillLinkModel.
+            get_question_skill_links_based_on_difficulty_equidistributed_by_skill( # pylint: disable=line-too-long
+                3, ['skill_id1'], 0.6
+            )
+        )
+        utils.random.seed(2)
+        question_skill_links_2 = (
+            question_models.QuestionSkillLinkModel.
+            get_question_skill_links_based_on_difficulty_equidistributed_by_skill( # pylint: disable=line-too-long
+                3, ['skill_id1'], 0.6
+            )
+        )
+        self.assertEqual(len(question_skill_links_1), 3)
+        self.assertEqual(len(question_skill_links_2), 3)
+        self.assertNotEqual(question_skill_links_1, question_skill_links_2)
```

Comment:
While this is ensuring the two aren't equal, it would be good to also verify the contents of question_skill_links_1 and question_skill_links_2

Reply:
The contents of question_skill_links_1 and question_skill_links_2 might be partially same so a one to one check might not be possible so I am not sure what you mean by verify the contents.

Reply:
So question_skill_links_1 and question_skill_links_2 would contain 3 out of the 5 models you have created above right?
What I am trying to say is do an assertItemsEqual(question_skill_links_1, [all the models that question_skill_links_1 should contain])
Same thing for question_skill_links_2

Reply:
Done.